demos/app/iot_demo.c

```diff
@@ -118,19 +118,63 @@ int main( int argc,
     /* For a secured connection, set the members of the credentials. */
     if( demoArguments.securedConnection == true )
     {
-        /* Set credential paths. */
+        /* Set credential information. */
         credentials.pClientCert = demoArguments.pClientCertPath;
         credentials.pPrivateKey = demoArguments.pPrivateKeyPath;
         credentials.pRootCa = demoArguments.pRootCaPath;
+        credentials.pUserName = NULL;
+        credentials.pPassword = NULL;
+
+        /* Set the MQTT username, as long as it's not empty or NULL. */
+        if( demoArguments.pUserName != NULL )
+        {
+            credentials.userNameSize = strlen( demoArguments.pUserName );
+
+            /* Check for compliance with the MQTT standard. */
+            if( credentials.userNameSize > USHRT_MAX )
+            {
+                IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+            }
+
+            if( credentials.userNameSize > 0 )
+            {
+                credentials.pUserName = demoArguments.pUserName;
+            }
+        }
+
+        /* Set the MQTT password, as long as it's not empty or NULL. */
+        if( demoArguments.pPassword != NULL )
+        {
+            credentials.passwordSize = strlen( demoArguments.pPassword );
+
+            /* Check for compliance with the MQTT standard. */
+            if( credentials.passwordSize > USHRT_MAX )
+            {
+                IOT_SET_AND_GOTO_CLEANUP( EXIT_FAILURE );
+            }
+
+            if( credentials.passwordSize > 0 )
+            {
+                credentials.pPassword = demoArguments.pPassword;
+            }
+        }
 
         /* By default, the credential initializer enables ALPN with AWS IoT,
-         * which only works over port 443. Disable ALPN if another port is
+         * which only works over port 443. Clear that value if another port is
          * used. */
         if( demoArguments.port != 443 )
         {
             credentials.pAlpnProtos = NULL;
         }
 
+        /* Per IANA standard:
+         * https://www.iana.org/assignments/tls-extensiontype-values/tls-extensiontype-values.xhtml. */
+        if( ( demoArguments.pUserName != NULL ) &&
+            ( demoArguments.awsIotMqttMode == true ) )
+        {
+            credentials.pAlpnProtos = "mqtt";
```

Comment:
Is the value of credentials.pAlpnProtos need to be "mqtt" even if demoArguments.port is 443?

Reply:
Yes, the proposal is to adopt the official application protocol string for MQTT as per IANA standard. Depending on the MQTT broker, and the gateway that sits in front of it, the app protocol specification may be required regardless of the port.

Reply:
Thanks. Is this already supported in AWS IoT MQTT Broker? The documentation I was referring to was specific to AWS IoT MQTT broker